var/plugins/barmanager.py

```diff
@@ -1100,6 +1109,158 @@ def hGetLastVote(source, user, params, checkOnly):
+        user, ','.join(params)), DBGLEVEL)
+
+    try:
+        return genericNoParameterCommand(source, user, params, checkOnly, "resetchevlevels")
+    except Exception as e:
+        spads.slog("Unhandled exception: " + str(sys.exc_info()
+                   [0]) + "\n" + str(traceback.format_exc()), 0)
+
+def hRename(source, user, params, checkOnly):
+    spads.slog("User %s called command rename with parameter(s) \"%s\"" % (
+        user, ','.join(params)), DBGLEVEL)
+
+    try:
+        return genericStringCommand(source, user, params, checkOnly, "rename", "^[a-zA-Z0-9_\-\[\] \<\>\+\|:]+$")
+    except Exception as e:
+        spads.slog("Unhandled exception: " + str(sys.exc_info()
+                   [0]) + "\n" + str(traceback.format_exc()), 0)
+
+def hWelcomeMessage(source, user, params, checkOnly):
+    spads.slog("User %s called command welcome-message with parameter(s) \"%s\"" % (
+        user, ','.join(params)), DBGLEVEL)
+
+    try:
+        return genericStringCommand(source, user, params, checkOnly, "welcome-message", "^.*$")
+    except Exception as e:
+        spads.slog("Unhandled exception: " + str(sys.exc_info()
+                   [0]) + "\n" + str(traceback.format_exc()), 0)
+
+def genericNoParameterCommand(source, user, params, checkOnly, cmd):
+    user = spads.fix_string(user)
+    for i in range(len(params)):
+        params[i] = spads.fix_string(params[i])
+
+    if len(params) > 0:
+        spads.slog(cmd + ": syntax error: more than 0 parameters", DBGLEVEL)
+        spads.sayPrivate(user, cmd + ": Too many parameters.")
+        return False
+
+    # All parameter checking was successful, return now if no action is desired
+    if checkOnly:
+        return True
+
+    spads.queueLobbyCommand(["SAYBATTLE", "$%" + cmd])
+
+def genericSingleIntegerCommand(source, user, params, checkOnly, cmd, defaultValue, minValue, maxValue):
+    user = spads.fix_string(user)
+    for i in range(len(params)):
+        params[i] = spads.fix_string(params[i])
+
+    if len(params) > 1:
+        spads.slog(cmd + ": syntax error: more than 1 parameter", DBGLEVEL)
+        spads.sayPrivate(user, cmd + ": Too many parameters.")
+        return False
+
+    newValue = None
+
+    if len(params) == 0:
+        newValue = defaultValue
+
+    if len(params) == 1:
+        newValue = int(params[0]) if params[0].isdecimal() else None
+
+    if newValue is None:
+        spads.slog(cmd + ": value error: param 1 is not numeric", DBGLEVEL)
+        spads.sayPrivate(user, cmd + ": Parameter is not numeric")
+        return False
+
+    # All parameter checking was successful, return now if no action is desired
+    if checkOnly:
+        return True
+
+    if newValue < minValue:
+        newValue = minValue
+    if newValue > maxValue:
+        newValue = maxValue
+
+    spads.queueLobbyCommand(["SAYBATTLE", "$%" + cmd + " " + str(newValue)])
+
+def genericStringCommand(source, user, params, checkOnly, cmd, validRegex):
```

Comment:
nit: `validRegex` can be made instance of the `Pattern`, so you pass to function e.g. `re.compile("^.*$")`, this way only correct regexp can be passed for sure, code will blow up potentially earlier not at handler call time, especially with my other comment about using command generator function, and also you do less regexp compilation.